docs/weave/weave-api-spec/04-querying.md

```diff
@@ -0,0 +1,143 @@
+
+- Path: `/wallets/name`, Data: `4A6F686E` (raw: `John`): wallets.Index("name").Get("4A6F686E")
+  - `wallets` are queried for the account with name `John`.
+
+### Multikey indexes
+
+There might be some cases where one index has multiple values. Multikeys exists for this purpose. Query response will be multiple values instead of one.
+
+### Prefixes
+
+There might be cases where all the data with has the same index prefix wanted to be retrieved. For this purpose you can do prefix searches using `?prefix` endpoints
+
+- Queries that made using prefix without a data field would result as listing all the objects saved under the bucket
+
+  - E.g. query below will return all the wallets saved under wallet bucket in [ResultSet](#Responses) response format
+
+    ```bash
+    curl -X POST -d '{ "json-rpc": 2.0, "id": "foobar321", "method": "abci_query",
+    "params": { "path": "/wallets?prefix", "data": "" } }' \
+    https://bns.kissnet.iov.one/
+    ```
+
+  - E.g. query below will return all the tokens saved under token bucket in again ResultSet.
+
+    ```bash
+      curl -X POST -d '{ "json-rpc": 2.0, "id": "foobar321", "method": "abci_query",
+      "params": { "path": "/tokens?prefix", "data": "" } }' \
+      https://bns.kissnet.iov.one/
+    ```
+
+- Path: `/wallets?prefix`, Data: `0123456789` (hex) -> db.Iterator(`0123456789`, `012345678A`)
+  - `wallets` are queried for the accounts starting with from `0123456789` to `012345678A`
+
+## Responses
+
+Since Weave queries routed to tendermint ABCI protocol, tendermint enforces responses to be in `key/value` format. Reference: [tendermint/abci-spec#query](https://tendermint.readthedocs.io/en/v0.21.0/abci-spec.html#query).
+
+```bash
+curl -X POST -d '{ "json-rpc": 2.0, "id": "foobar321", "method": "abci_query", "params": { "path": "/wallets?prefix", "data": "CBC76ADED2C9DB439DB4C8D714CF26" } }' https://bns.kissnet.iov.one/
+```
+
+When the curl command above executed, this response will be received:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": "foobar321",
+  "result": {
+    "response": {
+      "key": "ChljYXNoOsvHat7SydtDnbTI1xTPJtrlIpqR",
+      "value": "ChAKAggBEgoIlZrvOhoDSU9W",
+      "height": "529"
+    }
+  }
+}
+```
+
+### Empty result or not found
+
+For non-existent objects Weave returns the current block height. Such as:
+
+```bash
+curl -X POST -d '{ "json-rpc": 2.0, "id": "foobar321",
+"method": "abci_query", "params": { "path": "/wallets?prefix", "data": "0123456789" } }' \
+https://bns.kissnet.iov.one/
+```
+
+The response will be:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": "foobar321",
+  "result": {
+    "response": {
+      "height": "527" <--- current block height
+    }
+  }
+}
+```
+
+Since the request is with prefix, meaning multiple values will be returned.
```

Comment:
"Since the request is with prefix, meaning multiple values will be returned."

replace with

"Since the request is with a prefix, multiple values will be returned."